client/src/js/components/bhDepotSelect.js

```diff
@@ -0,0 +1,30 @@
+angular.module('bhima.components')
+  .component('bhDepotSelect', {
+    bindings : {
+      onSelect : '&',
+    },
+    templateUrl  : 'modules/templates/bhDepotSelect.tmpl.html',
+    controller   : bhDepotSelectController,
+    controllerAs : '$ctrl',
```

Comment:
All components are automatically bound with `controllerAs: '$ctrl'`.  You do not need to specify it.  [For reference, here is the documentation for Angular's component() method](https://docs.angularjs.org/guide/component#comparison-between-directive-definition-and-component-definition).